core/types/rollup_cost.go

```diff
@@ -130,11 +211,38 @@ func extractL1GasParams(config *params.ChainConfig, time uint64, data []byte) (l
 	fscalar := new(big.Float).SetInt(scalar)             // legacy: format fee scalar as big Float
 	fdivisor := new(big.Float).SetUint64(1_000_000)      // 10**6, i.e. 6 decimals
 	feeScalar = new(big.Float).Quo(fscalar, fdivisor)
-	costFunc = newL1CostFunc(l1Basefee, overhead, scalar, config.IsRegolith(time))
+	costFunc = newL1CostFuncBedrockHelper(l1Basefee, overhead, scalar, config.IsRegolith(time))
+	return
+}
+
+// extractEcotoneL1GasParams extracts the gas parameters necessary to compute gas from L1 attribute
+// info calldata after the Ecotone upgrade, but not for the very first Ecotone block.
+func extractL1GasParamsEcotone(data []byte) (l1Basefee *big.Int, costFunc l1CostFunc, err error) {
+	if len(data) != 164 {
+		return nil, nil, fmt.Errorf("expected 164 L1 info bytes, got %d", len(data))
+	}
+	// data layout assumed for Ecotone:
+	// offset type varname
+	// 0      <selector>
+	// 4     uint32 _basefeeScalar
+	// 8     uint32 _blobBasefeeScalar
+	// 12    uint64 _sequenceNumber,
+	// 20    uint64 _timestamp,
+	// 28    uint64 _l1BlockNumber
```

Comment:
these are the other way around in the monorepo PR

Reply:
This is the correct order. I noticed this was flipped in one of the function comments of the L2 contract PR which I fixed earlier today.  Will check the others...